### ISSUE TITLE
video blob download randomly fails

### ISSUE THREAD

**Issue by SuperSandro2000** [iv-org/invidious#1224]
On my private instance blob video download sometimes fails for no apparent reason and instead of retrying invidious just loops forever with the following errors:

```
video.min.js?v=3f97beb:12 VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) The media could not be loaded, either because the server or network failed or because the format is not supported. vt
(anonymous) @ video.min.js?v=3f97beb:12
video.min.js?v=3f97beb:21 Uncaught DOMException: Failed to read the 'buffered' property from 'SourceBuffer': This SourceBuffer has been removed from the parent media source.
    at vc.value (https://invidious.example.com/js/video.min.js?v=3f97beb:21:97851)
    at SourceBuffer.onUpdateendCallback_ (https://invidious.example.com/js/video.min.js?v=3f97beb:21:97088)
video.min.js?v=3f97beb:21 Uncaught DOMException: Failed to execute 'appendBuffer' on 'SourceBuffer': This SourceBuffer has been removed from the parent media source.
    at vc.<anonymous> (https://invidious.example.com/js/video.min.js?v=3f97beb:21:97602)
    at vc.value (https://invidious.example.com/js/video.min.js?v=3f97beb:21:98748)
    at vc.value (https://invidious.example.com/js/video.min.js?v=3f97beb:21:98566)
    at vc.value (https://invidious.example.com/js/video.min.js?v=3f97beb:21:97434)
    at Dc.value (https://invidious.example.com/js/video.min.js?v=3f97beb:21:116046)
    at Dc.value (https://invidious.example.com/js/video.min.js?v=3f97beb:21:114305)
    at Dc.value (https://invidious.example.com/js/video.min.js?v=3f97beb:21:113409)
    at https://invidious.example.com/js/video.min.js?v=3f97beb:21:100410
    at https://invidious.example.com/js/video.min.js?v=3f97beb:21:102473
    at Object.callback (https://invidious.example.com/js/video.min.js?v=3f97beb:21:10484)
/api/v1/comments/PKB4cioGs98?format=html&hl=en-US&thin_mode=false:1 Failed to load resource: the server responded with a status of 500 ()
player.js?v=3f97beb:65 An error occured in the player, reloading...
```

A reload ~~fixes~~ does not fix it but it is annoying when playing playlists and after 10 to 20 videos it fails again.

After some more debugging I noticed that this only happens with videos that are not available in dash format which I use by default. If this is the case it should probably default to the next available lower quality.

**Comment by omarroth** [comment]
This is an issue when the player has multiple sources that are the same resolution but different codecs. Should be fixed as part of 1eca969cf6b4096789014619285c98d1def40ee3.

**Comment by TheFrenchGhosty** [comment]
I assume that this is fixed.
